homeassistant/components/glances/sensor.py

```diff
@@ -245,20 +279,13 @@ async def async_setup_entry(
     """Set up the Glances sensors."""
 
     coordinator: GlancesDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
-    entities = []
+    entities: list[GlancesSensor] = []
 
-    for sensor_type, sensors in coordinator.data.items():
+    for sensor_type, sensors in (items := coordinator.data.items()):
         if sensor_type in ["fs", "sensors", "raid"]:
-            for sensor_label, params in sensors.items():
-                for param in params:
-                    if sensor_description := SENSOR_TYPES.get((sensor_type, param)):
-                        entities.append(
-                            GlancesSensor(
-                                coordinator,
-                                sensor_description,
-                                sensor_label,
-                            )
-                        )
+            _add_custom_sensors(coordinator, entities, sensor_type, sensors)
+        elif sensor_type == "gpu":
```

Comment:
Wouldn't it be simpler to change the key formatting for gpu in python-glances-api and avoid having a special case for Gpu here ?
https://github.com/home-assistant-ecosystem/python-glances-api/blob/a73164393868d6c76421208953e771b9297209ed/glances_api/__init__.py#L185

Reply:
By the way I can submit the PR to python-glances-api if you want to got this way. I already have another PR open for `network` which will require a new version of the library.

Reply:
Thanks, yes please! I was being impatient and hacky, hoping to get the change in before I would have less time. Wrong mindset. It is obviously better to change the formatting in the package itself than add complexity here.
Thank you for your offer to add the PR in there, I am still very busy.

Reply:
Hi @fhoekstra , I have drafted a pull request here:
https://github.com/home-assistant-ecosystem/python-glances-api/pull/35

The proposed format for the GPU name is:
`f"{sensor['name']} (GPU {sensor['gpu_id']})"`
which translates to : 
```
    "gpu": {
        "NVIDIA GeForce RTX 4080 (GPU 0)": {
            "mem": 13.333489176233513,
            "proc": 12,
            "temperature": 38,
            "fan_speed": 30,
        },
        "NVIDIA GeForce RTX 3080 (GPU 1)": {
            "mem": 8.41064453125,
            "proc": 26,
            "temperature": 51,
            "fan_speed": 0,
        },
    },
```
If you can have a quick look, I'll submit it to the library maintainer.

Reply:
PR submitted to python-glances-api

Reply:
Version 6.0 is now available with the formatting change on GPU name

Reply:
Version 0.6.0 has been merged into dev branch

Reply:
Thanks for the quick action, I've rebased the branch. Also moving the icon definitions but have some trouble with my local dev setup. If I can't figure it out quickly, I could give you write access to the branch so I'm not blocking you any longer